packages/core/contracts/chainbridge/BeaconOracle.sol

```diff
@@ -14,7 +14,7 @@ import "../oracle/implementation/Constants.sol";
  * price requests from the DVM on Mainnet.
  */
 abstract contract BeaconOracle {
-    enum RequestState { NeverRequested, Requested, Resolved }
+    enum RequestState { NeverRequested, PendingRequest, Requested, PendingResolve, Resolved }
```

Comment:
I'm assuming this contract is depending on derived contracts to move it from PendingRequest to Requested and PendingResolve to Resolved? Is that correct?

Might be a little confusing that this state is actively managed by base and derived contracts, but I can't think of another way to do it without creating another storage variable outside of RequestState....

Reply:
I see what you're saying. Do you think this would be more clear with (1) better comments, (2) internal methods like `_validateRequest` and `_validatePublish` which just advance the state, or (3) something else?

Reply:
I could imagine 2 being effective, yeah. That way all the logic is somewhat contained in the base class.

Reply:
Done, this removed some code from the derived Sink/Source oracles which is nice